src/main/java/net/neoforged/neoforge/transfer/transaction/TransactionManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.transaction;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import java.util.ArrayList;
+import java.util.List;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Manager for handling opening new {@link Transaction Transactions} or querying status of a transaction chain in a given thread.
+ *
+ * @see Transaction
+ */
+public final class TransactionManager {
```

Comment:
Based on intention, this entire class should be `@ApiStatus.Internal`. Any according APIs should be done through the `Transaction` class. See comments from @covers1624 regarding the request for `Transaction.open()` and `Transaction.getCurrentTransaction()`